contributing/samples.md

```diff
@@ -0,0 +1,257 @@
+## Creating a new sample
+
+Samples live in `examples/source`. Pick one of the existing category folders. 
+
+If your sample can live in a single html file, create a new `*.html` in  one of the sample category folders:
+
+```shell
+$ vim examples/source/1.components/amp-awesome.html
+```
+
+If your sample requires a custom backend API, create a new folder which is going to contain both, your sample html file, which must be named `index.html` and your api implementation `api.js`.
+
+```shell
+$ mkdir examples/source/1.components/amp-awesome
+$ vim examples/source/1.components/amp-awesome/index.html
+$ vim examples/source/1.components/amp-awesome/api.js
+```
+
+If your sample, does not fit into one of the existing categories, please [create an issue](https://github.com/ampproject/docs/issues/new) first and ask for feedback.
+
+## Frontmatter
+
+Samples can define a YAML-based frontmatter. Here is a template to get you started:
+
+```
+<!---
+- author: your-github-user-name
+- formats
+  - websites
+--->
+```
+
+**Note the triple dash `<!---`!**. Also make sure to list all supported formats (websites,ads,email,stories).
+
+If your sample is specific to an AMP Format, please define it as well:
+
+```
+- formats
+  - websites
+```
+
+Please list all experimental components that your sample uses:
+
+```
+experiments:
+  - amp-autocomplete
+```
+
+Other supported flags are:
+
+```yaml
+- formats [default: websites,ads,email,stories]:
+  - websites
+  - ads
+  - email
+  - stories
+- validAmp [default: true]
+  - true
+  - false
+- draft [default: true]
+  - true
+  - false
+- tags [default: '']
+  - ads-analytics
+  - dynamic-content
+  - layout
+  - media
+  - presentation
+  - social
+  - personalization
+```
+
+## Writing the sample
+
+Samples are annotated HTML files. Use HTML comments (`<!-- ... -->`) to document your sample code:
```

Comment:
This sounds like we should wrap our sample code in HTML comments. Maybe something like:

Add documentation to your sample's code by wrapping text in HTML comments.

Reply:
done